File: src/offworldrecords.java
import java.util.Scanner;

class OffWorldRecords {
    public static void main(String[] args) {
        Scanner s = new Scanner(System.in);
        int n = s.nextInt();
        int c = s.nextInt();
        int p = s.nextInt();
        int ans = 0;

        for (int i = 0; i < n; i++) {
            int h = s.nextInt();
            if (h > c + p) {
                ans += 1;
                p = c;
                c = h;
            }
        }
        System.out.println(ans);
    }
}

File: src/offworldrecords.lua
n, c, p = io.read("*n", "*n", "*n")
ans = 0
for _ = 1, n do
    h = io.read("*n")
    if h > c + p then
        ans = ans + 1
        c, p = h, c
    end
end
print(ans)
